src/model/schema.js

```diff
@@ -465,7 +465,7 @@ export class MarkType extends SchemaItem {
 // with node type constructors and another similar object associating
 // mark names with mark type constructors.
 export class SchemaSpec {
-  // :: (?Object<NodeType>, ?Object<MarkType>)
+  // :: (?Object<SubClass<NodeType>>, ?Object<SubClass<MarkType>>)
```

Comment:
I'm using `kinds` as a static property, though that is mostly incidental. The `register` and `updateAttrs` methods are more important -- they allow user code to further configure a subclass.